docs/source/user_guide/timeseries.ipynb

```diff
@@ -328,6 +328,45 @@
     "plt.show()"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "It is desirable to enhance the decomposer component with a guess at the period of the seasonal aspect of the signal before decomposing it.  To do that, we can use the `determine_periodicity(X, y)` function of the `Decomposer` class."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "period = pdc.determine_periodicity(X_train_time, y_train_time)\n",
+    "print(period)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The children of the `Decomposer` class, if not set in the constructor, sets its `seasonal_period` parameter to -1.  If the `Decomposer` object is fit with a `seasonal_period` of -1, it will take on a default value of 7, which is good for daily data signals that have a known seasonal component period that is weekly.  \n",
+    "\n",
+    "However, if the seasonal period is not known before hand, the `set_seasonal_period()` convenience function will look at the target data, determine the period and set the `Decomposer` object appropriately."
```

Comment:
What do you think of this in the cell:

```
The children of the `Decomposer` class, if not explicitly set in the constructor, set their `seasonal_period` parameter
based on a `statsmodels` function, `freq_to_period`, that considers the frequency of the datetime data.  If the
`PolynomialDecomposer` object is fit with `seasonal_period` not explicitly set, it will take on a default value of 7,
which is good for daily data signals that have a known seasonal component period that is weekly.

However, if the seasonal period is not known beforehand, the `set_seasonal_period()` convenience function will look at the target data,
determine a better guess for the period and set the `Decomposer` object appropriately.
```

Reply:
That makes a bit more sense. I'm still slightly confused though, does the `set_seasonal_period()` function replace the `freq_to_period` function with a better guess, or are they completely unrelated?

Reply:
`set_seasonal_period()` does indeed set `seasonal_period` with a better guess.  `freq_to_period` basically just looks at the data and says "oh, the granularity of the time series is days, let's use 7." and has a similar mapping for other frequencies.  `set_seasonal_period()` will actually invoke the seasonal period detection to make a good guess, then set it.